internal/core/services/room_manager/room_manager.go

```diff
@@ -98,9 +98,36 @@ func (m *RoomManager) DeleteRoom(ctx context.Context, gameRoom *game_room.GameRo
 		return fmt.Errorf("failed to delete instance on the runtime: %w", err)
 	}
 
-	err = m.SetRoomStatus(ctx, gameRoom, game_room.GameStatusTerminating)
+	newGameRoom := *gameRoom
```

Comment:
Is there any specific reason for copying the game room? I think it is ok to have this side effect in this function.

Reply:
No specific reason, going to change it!

Reply:
786d396